src/plugins/gui/providers/bityProvider.ts

```diff
@@ -390,32 +396,45 @@ export const bityProvider: FiatProviderFactory = {
           bityCurrencies = asBityCurrencyResponse(result).currencies
         } catch (error: any) {
           console.error(error)
-          return allowedCurrencyCodes[direction]
+          return supportedAssets.getFiatProviderAssetMap({
+            direction,
+            region: `*`,
+            payment: supportedPaymentType
+          })
         }
+
         for (const currency of bityCurrencies) {
-          let isAddCurrencySuccess = false
           if (currency.tags.length === 1 && currency.tags[0] === 'fiat') {
-            allowedCurrencyCodes[direction].fiat['iso:' + currency.code.toUpperCase()] = currency
-            isAddCurrencySuccess = true
+            const fiatCurrencyCode = 'iso:' + currency.code.toUpperCase()
+            supportedAssets.add.direction(direction).region('*').fiat(fiatCurrencyCode).payment(supportedPaymentType)
```

Comment:
You're insistently using `*` for direction vs the actual passed in `direction`. seems like this will cause issues since you'll add the direction on the first call to `getSupportedAssets` but on a subsequent call for the opposite direction, you'll return false vs re-querying and filling in the cache